lib/libzfs/libzfs_util.c

```diff
@@ -452,6 +452,11 @@ zfs_standard_error_fmt(libzfs_handle_t *hdl, int error, const char *fmt, ...)
 		    "pool I/O is currently suspended"));
 		zfs_verror(hdl, EZFS_POOLUNAVAIL, fmt, ap);
 		break;
+	case ENOTDIR:
+		zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+		    "parent is not a filesystem"));
```

Comment:
I can live with this, but I think it would be better to add a new value to zfs_errno_t for this error.